components/connectivity-adapter/internal/appregistry/service/handler.go

```diff
@@ -1,37 +1,72 @@
 package service
 
 import (
-	"log"
+	"context"
+	"fmt"
 	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/gqlcli"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/reqerror"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
 )
 
-type Handler struct{}
+const serviceIDVarKey = "serviceId"
+
+type Handler struct {
+	cliProvider gqlcli.Provider
+	logger      *log.Logger
+}
 
-func NewHandler() *Handler {
-	return &Handler{}
+func NewHandler(cliProvider gqlcli.Provider, logger *log.Logger) *Handler {
+	return &Handler{
+		cliProvider: cliProvider,
+		logger:      logger,
+	}
 }
 
 func (h *Handler) Create(rw http.ResponseWriter, rq *http.Request) {
-	log.Println("Create")
+	h.logger.Println("Create")
 	rw.WriteHeader(http.StatusOK)
 }
 
 func (h *Handler) Get(rw http.ResponseWriter, rq *http.Request) {
-	log.Println("Get")
+	h.logger.Println("Get")
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h *Handler) Update(rw http.ResponseWriter, rq *http.Request) {
-	log.Println("Update")
+func (h *Handler) List(rw http.ResponseWriter, rq *http.Request) {
+	h.logger.Println("List")
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h *Handler) Delete(rw http.ResponseWriter, rq *http.Request) {
-	log.Println("Delete")
+func (h *Handler) Update(rw http.ResponseWriter, rq *http.Request) {
+	h.logger.Println("Update")
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h *Handler) List(rw http.ResponseWriter, rq *http.Request) {
-	log.Println("List")
-	rw.WriteHeader(http.StatusOK)
+func (h *Handler) Delete(writer http.ResponseWriter, request *http.Request) {
+	gqlCli := h.cliProvider.GQLClient(request)
+
+	vars := mux.Vars(request)
+
+	id := vars[serviceIDVarKey]
+	gqlRequest := prepareUnregisterApplicationRequest(id)
+
+	err := gqlCli.Run(context.Background(), gqlRequest, nil)
+	if err != nil {
+		if apperrors.IsNotFoundError(err) {
+			message := fmt.Sprintf("entity with ID %s not found", id)
+			reqerror.WriteErrorMessage(writer, message, apperrors.CodeNotFound)
+			return
+		}
+
+		h.logger.Error(errors.Wrapf(err, "while deleting service with ID %s", id))
```

Comment:
Just a minor general comment, when you create a log message this way and you have some parameter, which has a type of string, has an empty string value then it is hard to catch this in the log output. Maybe it would be more helpful to format the log output, something like `fmt.Sprintf("while deleting service [ID=%s]", id)`

Reply:
Hmm, ok, so I will use `WithFields` helper method from logrus - it will be cleaner solution for that